File: daily/2026/2130_Maximum_Twin_Sum_of_a_Linked_List.cpp
#include <iostream>
#include <vector>

using namespace std;

// Definition for singly-linked list.
struct ListNode {
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};

ListNode* createList(vector<int>& nodes){
    if (nodes.empty()) return nullptr;
    ListNode* head = new ListNode(nodes[0]);
    ListNode* ptr = head;
    int n = nodes.size();
    for (int i = 1; i < n; i++){
        ptr->next = new ListNode(nodes[i]);
        ptr = ptr->next;
    }
    return head;
}

class Solution {
public:
    int pairSum(ListNode* head) {
        vector<int> v;
        ListNode* ptr = head;
        while(ptr){
            v.push_back(ptr->val);
            ptr = ptr->next;
        }
        int n = v.size();
        int ans = 0;
        for (int i = 0; i < n / 2; i++){
            ans = max(ans, v[i] + v[n - 1 - i]);
        }
        return ans;
    }
};

int main(){
    vector<int> nodes = {5,4,2,1};
    // vector<int> nodes = {4,2,2,3};
    // vector<int> nodes = {1,100000};
    ListNode* head = createList(nodes);
    Solution S;
    cout << S.pairSum(head) << endl;
    return 0;
}

File: daily_python/2130_Maximum_Twin_Sum_of_a_Linked_List.py
import os
import sys
from typing import List, Optional

# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class L:
    def __init__(self, nodes: Optional[List[int]]):
        self.head = None if not nodes else ListNode(nodes[0])
        ptr = self.head
        for i in range(1, len(nodes)):
            ptr.next = ListNode(nodes[i])
            ptr = ptr.next

class Solution:
    def pairSum(self, head: Optional[ListNode]) -> int:
        v = []
        ptr = head
        while ptr:
            v.append(ptr.val)
            ptr = ptr.next
        n = len(v)
        ans = 0
        for i in range(n // 2):
            ans = max(ans, v[i] + v[n-1-i])
        return ans
    
def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            nodes = list(map(int, lines[0].split('=')[1].strip()[1:-2].split(',')))
            dataset.append(nodes)
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for nodes in dataset:
        results.append(solution.pairSum(L(nodes).head))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')